libs/src/api/user.js

```diff
@@ -545,14 +545,20 @@ class Users extends Base {
       // Update user creator_node_endpoint on chain if applicable
       if (newMetadata.creator_node_endpoint !== oldMetadata.creator_node_endpoint) {
         phase = phases.UPDATE_CONTENT_NODE_ENDPOINT_ON_CHAIN
-        await this.contracts.UserFactoryClient.updateCreatorNodeEndpoint(userId, newMetadata['creator_node_endpoint'])
-        // Ensure DN has indexed creator_node_endpoint change
-        await this._waitForCreatorNodeEndpointIndexing(userId, newMetadata.creator_node_endpoint)
+        if (firstTime) {
+          this.contracts.UserFactoryClient.updateCreatorNodeEndpoint(userId, newMetadata['creator_node_endpoint'])
```

Comment:
is pulling out the await here intentional? shouldn't we wait for the endpoint to be updated, and then upload metadata?

Reply:
it was intentional -- but i think it probably doesn't make much of a diff if we await it (and it would let us retry if it fails). key is not waiting for the indexing part at all and letting the ensureprimary part be skipped.
good callout